src/plugins/graphql.js

```diff
@@ -52,24 +52,37 @@ function createWrapParse () {
   }
 }
 
-function wrapFields (fields, tracer, config, responsePathAsArray) {
-  Object.keys(fields).forEach(key => {
-    const field = fields[key]
+function wrapFields (type, tracer, config, responsePathAsArray) {
 
-    if (typeof field.resolve === 'function') {
+  if (type._datadog_patched) {
+    return
+  }
+
+  type._datadog_patched = true
+
+  Object.keys(type._fields).forEach(key => {
+    const field = type._fields[key]
+
+    if (typeof field.resolve === 'function' && !field.resolve._datadog_patched) {
       field.resolve = wrapResolve(field.resolve, tracer, config, responsePathAsArray)
     }
 
-    if (field.type && field.type._fields) {
-      wrapFields(field.type._fields, tracer, config, responsePathAsArray)
+
+    if (field.type && !field.type._datadog_patched) {
```

Comment:
The `_datadog_patched` property should be on the resolve function directly. This will avoid the same issue when a function is shared between 2 types.

Reply:
Yeah, _datadog_patched would never get set on something like a list so is a pointless check here. Though isn't it needed on the type to stop recursion.

Reply:
I added a recursive type to the test schema, so because the tests are passing its working. But a dedicated test would be much better.